### ISSUE TITLE
Caching writes?

### ISSUE THREAD

**Issue by c-prather** [parthenon-hpc-lab/parthenon#783]
There is an HDF5 addition (virtual object layer, or VOL) to cache writes (and, though less useful, reads) on local flash storage of each node, allowing them to be copied back to slower network storage while execution continues.

Repository is [here](https://github.com/hpc-io/vol-cache)

Some clusters (Summit, Delta) have alternative solutions to this problem using flash cache at a filesystem level, but a VOL-level solution would enable the same thing on pretty much any cluster.

**Comment by Yurlungur** [comment]
I don't like depending on this automatically---i.e., we should be able to run without it. But it would be nice to run with it when available.

**Comment by c-prather** [comment]
Absolutely, it should be optional, even when using HDF5.  I think the actual code changes they list in the README are small enough that it would be an easy compile-time flag to maintain, and my hope would be that the repository could just be a submodule or something, with an (optional) Parthenon-managed build.

**Comment by Yurlungur** [comment]
:+1: That would be super nice! I'd be interested in such a capability.

**Comment by pgrete** [comment]
In principle, this sounds like a nice solution.
From a practical point of view, I'm wondering how strongly we need this kind of feature.
So far I've been quite happy with the HDF5 performance given that we typically write comparatively large chunks of data per rank (even for AMR sims) versus lots of tiny writes (where caches are really helping a lot).
I'm just bringing this up as it's something we'd need to test/maintain.
